juriscraper/opinions/united_states/territories/guam.py

```diff
@@ -17,50 +23,90 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.court_id = self.__module__
         self.url = "https://guamcourts.org/Supreme-Court-Opinions/Supreme-Court-Opinions.asp"
+        self.status = "Published"
+
         self._year = date.today().year
         self.parameters = {"Year": str(self._year)}
         self.method = "POST"
-        self.status = "Published"
 
         # The year dropdown goes back to 1990, but the Court wasn't
         # created until 1996 and there are no opinions posted for
         # prior years.
         self.back_scrape_iterable = range(1996, self._year)
 
-    def _process_html(self):
-        year = 2019 if self.test_mode_enabled() else self._year
-        middle_of_year = "July 2, %d" % year
+    def _process_html(self) -> None:
+        """Process HTML into case objects
+
+        Some docket numbers are a consolidation of other dockets
+        For example: "CVA12-018 (consolidated with CVA12-030)"
+        Deleting the date and citation from the free text allows us
+        to catch these dockets name
 
-        for table in self.html.xpath(
-            '//a[@id="Opinion"]/following-sibling::table'
-        ):
+        :return: None
+        """
+        row_xpath = '//a[@id="Opinion"]/following-sibling::table'
+        for table in self.html.xpath(row_xpath):
             text = table.xpath(".//td/text()")[0]
-            # Text is typically something like this:
-            # CVA16-016, 2018 Guam 8, July 26, 2018
-            # But formatting varies from year to year, and some elements
-            # are not provided for some cases.
 
-            date_match = re.search(r"[A-Za-z]+\.?\s+[0-9]+,\s+[0-9]+", text)
-            docket_match = re.search(r"[A-Z]+[0-9]+\-+[0-9]+", text)
-            neutral_citation_match = re.search(r"[0-9]+\s+Guam\s+[0-9]+", text)
+            date = self.find_date(text)
+            if not date:
+                logger.error(
+                    "Could not find date for record in guam scraper Year: %s",
+                    self._year,
+                )
+                continue
```

Comment:
do you know about `date_filed_is_approximate` 

If we cant find the date, but we can find the year and month we usually assign the date to the 15th of the month and year and set date_filed_is_approximate to True, 

if we only find the year - we can do the same - but assign the date to the middle of the year and state if date_filed_is_approximate = True again

Reply:
I have updated the PR using `date_filed_is_approximate`, please check again